src/astro/databases/snowflake.py

```diff
@@ -57,6 +81,322 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Checks if the table and schema exists on Snowflake.
+        If table and schema exists, create stage based on integration passed by user.
+        Generate query and run COPY INTO command and return True else returns False
+
+        :param source_file: Source file
+        :param target_table: Output target table on snowflake
+        :param chunk_size: Chunk size for the file
+        :param if_exists: Update strategy for file
+        :param kwargs:
+        :return: None
+        """
+
+        # Create new table from file if snowflake table does not exists
+        if if_exists == "replace" and not self.table_exists(table=target_table):
+            self.drop_table(target_table)
+            self.create_emtpy_table(
+                source_file=source_file, target_table=target_table, if_exists=if_exists
+            )
+
+        # check if snowflake table exists
+        if self.table_exists(table=target_table):
+            logging.info("Table exists.")
+            stage_name = self.create_stage(source_file=source_file)
+            sql_parts = self.create_copy_into_sql_parts(
+                source_file=source_file, target_table=target_table, stage=stage_name
+            )
+            copy_query = "\n".join(sql_parts)
+
+            logging.info("Executing COPY command...")
+            execution_info = self.hook.run(
+                copy_query, autocommit=kwargs.get("autocommit", True)
+            )
+            logging.info("COPY command completed")
+            logging.info(execution_info)
+            self.drop_stage(stage_name=stage_name)
+
+    def create_stage(self, source_file: File) -> str:
+        """
+        Creates a stage name based on source_file path and storage integration provided.
+
+        @param source_file: Source file
+        @return: Stage name created
+        """
+        # Here,
+        # To copy dataset from gcs_path = gs://sample-bucket/workspace/sample.csv" to snowflake table
+        # where stage is WORKSPACE_STAGE, following steps has to be completed.
+
+        # The path of stage should be to the folder the file is in. Command to create STAGE would look
+        # like following:
+        # CREATE OR REPLACE STAGE <warehouse_name>.WORKSPACE_STAGE URL='gcs://sample-bucket/workspace'
+        # FILE_FORMAT=(TYPE=CSV, TRIM_SPACE=TRUE) COPY_OPTIONS=(MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE)
+        # storage_integration = <storage_integration_name>;
+
+        # COPY INTO query would be:
+        # COPY INTO sample_table from '@WORKSPACE_STAGE/sample.csv';
+        FILE_FORMAT_MAPPING = {
+            FileType.CSV: "CSV",
+            FileType.PARQUET: "PARQUET",
+            FileType.NDJSON: "JSON",
+        }
+        snowflake_query_file_storage_mapping = {"gs": "gcs", "s3": "s3"}
+        stage_name = generate_stage_name()
```

Comment:
The `stage_name` needs to be the qualified name, otherwise, Snowflake raises an exception.